app/scripts/lib/account-tracker.js

```diff
@@ -586,25 +593,36 @@ export default class AccountTracker {
     try {
       const balances = await ethContract.balances(addresses, ethBalance);
 
+      const accounts = this.#getAccountsForChainId(chainId);
+      const newAccounts = {};
+      Object.keys(accounts).forEach((address) => {
+        if (!addresses.includes(address)) {
+          newAccounts[address] = { address, balance: null };
+        }
+      });
       addresses.forEach((address, index) => {
         const balance = balances[index] ? balances[index].toHexString() : '0x0';
         newAccounts[address] = { address, balance };
       });
 
       const { accountsByChainId } = this.store.getState();
-      accountsByChainId[chainId] = newAccounts;
-      this.store.updateState({ accountsByChainId });
-      if (chainId === this.getCurrentChainId()) {
-        this.store.updateState({ accounts: newAccounts });
-      }
+      this.store.updateState({
+        ...(chainId === this.getCurrentChainId() && {
+          accounts: newAccounts,
+        }),
+        accountsByChainId: {
+          ...accountsByChainId,
+          [chainId]: newAccounts,
+        },
+      });
     } catch (error) {
       log.warn(
         `MetaMask - Account Tracker single call balance fetch failed`,
         error,
       );
       Promise.all(
```

Comment:
Another chance here to revisit whether this should be Promise.all vs Promise.allSettled. I think the latter. @shanejonas agrees: https://github.com/MetaMask/metamask-extension/pull/21868#discussion_r1412386135

Reply:
cc @mcmire @Gudahtt any thoughts?

Reply:
Yes, that's a good call.

Reply:
done https://github.com/MetaMask/metamask-extension/pull/22217/commits/e4a781096b6d36d67b8f15a9567ec190e3ba3dd5